numpy/random/_generator.pyx

```diff
@@ -4162,7 +4216,127 @@ cdef class Generator:
 
         return diric
 
-    # Shuffling and permutations:
+    def randomly_permute(self, object x, axis=None):
+        """
+        Randomly permute `x` along axis `axis` in-place.
+
+        If `axis` is None, the flattened array is shuffled.
+
+        Unlike `shuffle`, each slice along the given axis is shuffled
+        independently of the others.
+
+        Parameters
+        ----------
+        x : ndarray
+            Array to be randomly permuted.
+        axis : int, optional
```

Comment:
Should these support tuple[int] as well, e.g.,

```
gen = default_rng()
x = gen.standard_normal(5,6,7)
gen.randomly_permute(x, axis=(0,1)):
```

which would shuffle the element for each pair of the first 2 axes.

Reply:
That could be added (but I haven't looked into how hard it would be).  I'm not sure how to interpret "which would shuffle the element for each pair of the first 2 axes."  I would expect that example to shuffle the elements within the 2-d slices. That is, after the call, `x[:, :, 0]` contains the same elements as before the call, but shuffled, and the shuffle is applied to the flattened slice (so any given element could end up anywhere in the 2-d slice). Likewise for `x[:,:,1]`, etc.

Reply:
I agree that is the correct generalization, but am happy to only support `None` and a single int for now (with `None` effectively being the same as `range(ndim)` here).

Reply:
OK, for now I'll stick with None or a single integer, and leave handling of a tuple as a future enhancement.